pkg/scalers/metrics_api_scaler.go

```diff
@@ -210,13 +210,13 @@ func (s *metricsAPIScaler) getMetricValue(ctx context.Context) (*resource.Quanti
 	if err != nil {
 		return nil, err
 	}
+	defer r.Body.Close()
 
 	if r.StatusCode != http.StatusOK {
-		msg := fmt.Sprintf("api returned %d", r.StatusCode)
+		msg := fmt.Sprintf("%s: api returned %d", r.Request.URL.String(), r.StatusCode)
```

Comment:
Looking good, the only concern is, that we might be exposing confidental info. Might be worth to mask the url?
https://github.com/kedacore/keda/blob/877d0f6425452c4e2bc9bca8f21a7c48f386a26c/pkg/util/normalize_string.go#L43

Reply:
Might be a good idea, but how do I determine which info is confidential?

Reply:
we can use the referenced function (or modify it a little bit) to mask domain and subdomain and keep the path.

Reply:
If we need only path, we can just go with r.Request.URL.Path instead of full URL

Reply:
yeah, that's even better